composables/useErc725.ts

```diff
@@ -39,17 +44,211 @@ const fetchProfile = async (profileAddress: Address): Promise<Profile> => {
+          data = await fetchLSP8Assets(address, profile.address)
+
+          const assets =
+            data?.map(asset => ({
+              address,
+              standard,
+              data: asset,
+            })) || []
+
+          return assets
+        }
+        case 'LSP7DigitalAsset': {
+          assertAddress(profile.address)
+          data = await fetchLSP7Assets(address, profile.address)
+        }
+        default:
+          return [
+            {
+              address,
+              standard,
+              data,
+            },
+          ]
+      }
+    })
+  )
+
+  return (await assets).flat() as SupportedAssets[]
+}
+
+const fetchLSP4Metadata = async (
+  assetAddress: Address
+): Promise<[string, string, LSP4DigitalAssetJSON]> => {
+  const erc725 = getInstance(
+    assetAddress,
+    LSP4DigitalAsset as ERC725JSONSchema[]
+  )
+
+  try {
+    const lsp4DigitalAsset = await erc725.fetchData([
+      'LSP4TokenName',
+      'LSP4TokenSymbol',
+      'LSP4Metadata',
+    ])
+    const LSP4TokenName =
+      typeof lsp4DigitalAsset[0]?.value === 'string'
+        ? lsp4DigitalAsset[0]?.value
+        : ''
+    const LSP4TokenSymbol =
+      typeof lsp4DigitalAsset[1]?.value == 'string'
+        ? lsp4DigitalAsset[1]?.value
+        : ''
+    const LSP4Metadata = validateLSP4MetaData(
+      lsp4DigitalAsset[2].value as URLDataWithHash
+    )
+    return [LSP4TokenName, LSP4TokenSymbol, LSP4Metadata]
+  } catch (error) {
+    console.log(error)
+    return [
+      '',
+      '',
+      {
+        LSP4Metadata: {
+          description: '',
+          links: [],
+          images: [[]],
+          icon: [],
+          assets: [],
+        },
+      },
+    ]
+  }
+}
+
+const fetchLSP8Metadata = async (
+  tokenId: string,
+  assetAddress: Address
+): Promise<LSP4DigitalAssetJSON> => {
+  const lsp8MetadataGetter = async (
+    tokenIdType: string,
+    tokenId: string
+  ): Promise<LSP4DigitalAssetJSON> => {
+    const lsp8Metadata = await erc725.fetchData([
+      {
+        keyName: `LSP8MetadataJSON:<${tokenIdType}>`,
+        dynamicKeyParts: tokenId,
+      },
+    ])
+    return validateLSP4MetaData(lsp8Metadata[0].value)
+  }
+
+  const erc725 = getInstance(
+    assetAddress,
+    LSP8IdentifiableDigitalAsset as ERC725JSONSchema[]
+  )
+
+  try {
+    const lsp8DigitalAsset = await erc725.fetchData(['LSP8TokenIdType'])
+    const tokenIdType = lsp8DigitalAsset[0].value.toString()
+
+    //fetch LSP8MetadataJSON depending on tokenIdType
+    if (tokenIdType === TokenIdType.address) {
+      return lsp8MetadataGetter(
+        'address',
+        // ethers.utils.hexDataSlice(tokenId.toString(), 12)
+        tokenId.toString()
+      )
+    } else if (tokenIdType === TokenIdType.number) {
+      return lsp8MetadataGetter('uint256', parseInt(tokenId).toString())
+    } else if (tokenIdType === TokenIdType.bytes32) {
+      return lsp8MetadataGetter('bytes32', tokenId.toString())
+    } else {
```

Comment:
it looks like a good candidate for a switch case on the `tokenIdType` variable no?